components/progress-ring/index.jsx

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
+/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */
+
+// Implements the [Progress Ring design pattern](https://lightningdesignsystem.com/components/progress-ring/) in React.
+// Based on SLDS v2.4.5
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import { PROGRESS_RING } from '../../utilities/constants';
+import Icon from '../icon';
+import ProgressRingShape from './ring-shape';
+
+/**
+ * The themes available for the progress ring
+ */
+export const THEME_OPTIONS = Object.freeze({
+	WARNING: 'warning',
+	EXPIRED: 'expired',
+	COMPLETE: 'complete'
+});
+
+/**
+ * The CSS classes associated with each theme
+ */
+const THEME_CLASSES = {
+	[THEME_OPTIONS.COMPLETE]: 'slds-progress-ring_complete',
+	[THEME_OPTIONS.WARNING]: 'slds-progress-ring_warning',
+	[THEME_OPTIONS.EXPIRED]: 'slds-progress-ring_expired'
+};
+
+const propTypes = {
+	/**
+	 * HTML id for component.
+	 */
+	id: PropTypes.string,
+	/**
+	 * CSS class names to be added to the container element. `array`, `object`, or `string` are accepted.
+	 */
+	className: PropTypes.oneOfType([PropTypes.array, PropTypes.object, PropTypes.string]),
+	/**
+	 * The theme applied to the ring.
+	 */
+	theme: PropTypes.oneOf([THEME_OPTIONS.WARNING, THEME_OPTIONS.EXPIRED, THEME_OPTIONS.COMPLETE]),
+	/**
+	 * Overrides the icon to be displayed.
+	 */
+	icon: PropTypes.node,
+	/**
+	 * Display the icon associated with the theme.
+	 */
+	showIcon: PropTypes.bool,
+	/**
+	 * Percentage of progress completion, ranging [0, 100].
+	 */
+	value: PropTypes.number.isRequired
+};
+
+const defaultProps = {
+	showIcon: false
```

Comment:
The no-default-false-values seems to be advocating using double negative but the ES lint rules do not allow it. `if(!!myValue)` will cause ES Lint to fail because of rule `no-extra-boolean-cast`

Reply:
Let's add `no-extra-boolean-cast: 0` to https://github.com/salesforce/design-system-react/blob/master/.eslintrc for now and do the double cast.

Reply:
I'll update the eslint sometime. As it is now is good.